website/static/js/filepage/index.js

```diff
@@ -194,6 +195,49 @@ var FileViewPage = {
                 ], '.')
             ]);
         }
+        var sharePopover =  {view: function() { return m('button.btn.btn-sm.btn-primary.file-share', {onclick: function () {
```

Comment:
This component only needs to get declared once; no need to re-declare every time `FileViewPage.view`  gets called (which may be many times).

Instead, declare it at the top level of the module:

```js
var SharePopover = {
    view: function() {...}
}

var FileViewPage = {...}
```